fabric-datafixer-v1/src/main/java/net/fabricmc/fabric/api/datafixer/v1/TypeReferenceHelper.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datafixer.v1;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import com.mojang.datafixers.DSL;
+import com.mojang.datafixers.schemas.Schema;
+import com.mojang.datafixers.types.templates.TypeTemplate;
+
+import net.fabricmc.fabric.impl.datafixer.TypeRefHelperImpl;
+
+/**
+ * A helper which deals with registering Entities and BlockEntities along with registering their custom elements to be fixed.
+ *
+ * <p>This should be used inside of a {@link DataFixerEntrypoint} when registering an (block)entity or TypeReferences, otherwise issues occur with other DataFixers having no access your (block)entities, causing some DataFixes to fail.
+ * These types only apply the singular transformation, so any custom elements will have to be added manually via {@link DSL#fields(String, TypeTemplate)} or {@link DSL#optionalFields(String, TypeTemplate)}
+ */
+public interface TypeReferenceHelper {
+	/**
+	 * Gets the instance of the {@link TypeReferenceHelper}.
+	 */
+	TypeReferenceHelper HELPER = TypeRefHelperImpl.INSTANCE;
+
+	/**
+	 * Registers a type with a inTile element.
```

Comment:
Hmm, what is in tile?

Reply:
inTile had something to do with the arrow entities and tridents, maybe it's the state of the block the object is within or something similar?

Reply:
mind describe this more clearly in the jd?

Reply:
Is this like the `inBlockState` in the projectile NBT?
![nbt format on mcwiki](https://user-images.githubusercontent.com/6596629/69649095-2d924480-1075-11ea-9673-b6c0cd3618e1.png)